### PR TITLE
[REVIEW] Reduce number of build archs

### LINKED ISSUE #244: [BUG] Reduce number of architectures in build process
Match architectures used in RapidsAI default build process.

Change minimum build architecture from 3.5 to 6.0.

### PR DESCRIPTION
**Pull request by mnicely**
Closes #244 

### Info
The PR makes the minimum build architecture CC 50. 

Currently the other libraries default to compute capabilities `60,61,62,70,72,75`, plus `80` if CUDA 11 active.

cuSignal will leave in Maxwell architecture natively to support our large Jetson user base. 

This PR also fixes a bug where if `--allgpuarch` was not passed, all possible architectures were built. 

Now, if `--allgpuarch` is not passed, we check if `CUDA_VISIBLE_DEVICES` is set. 
If yes, build those architectures.
If no, build architectures for all installed devices.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by awthomp — CHANGES_REQUESTED** [review]
@mnicely and I spoke about this offline, but conda fatbins will only include GPU architecture that includes Pascal and newer GPUs. Users on older GPU architectures will need to build these fatbins from source. Matt is going to update the documentation to make this clearer.

File: README.md [hunk 1]
```diff
@@ -115,6 +115,12 @@ The complete cuSignal API documentation including a complete list of functionali
 [cuSignal 0.15 API](https://docs.rapids.ai/api/cusignal/stable/) | [cuSignal 0.16 Nightly](https://docs.rapids.ai/api/cusignal/nightly/)
 
 ## Installation
+### Note: cuSignal installs through Conda will run natively on **ALL** supported GPU architectures.
+
+1. CUDA 10.X -- [60, 61, 62] (Pascal), [70, 72] (Volta), [75] (Turing)
+2. CUDA 11.X -- [60, 61, 62] (Pascal), [70, 72] (Volta), [75] (Turing), [80] (Ampere)
+
+If you need to run on an older compute capability such as Kepler (CC 3.X) or Maxwell (CC 5.X), you will need to build from source. Target architecture will be automatically detected at build time.
```
Comment (awthomp) [inline README.md:123]: To be clear, will the _entire_ cusignal library need to be installed from source or will users simply need to run `./build.sh` to compile the fatbins alone?

Comment (awthomp) [reply README.md:123]: Let's add Maxwell back into the source conda build by default and remove these 4 lines of documentation.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by mnicely** [comment]
@awthomp Can you re-review? I'll push this in the morning if you're happy with it.

File: README.md [hunk 2]
```diff
@@ -115,6 +115,10 @@ The complete cuSignal API documentation including a complete list of functionali
 [cuSignal 0.15 API](https://docs.rapids.ai/api/cusignal/stable/) | [cuSignal 0.16 Nightly](https://docs.rapids.ai/api/cusignal/nightly/)
 
 ## Installation
+### Note: cuSignal installs through Conda will run natively on **ALL** supported GPU architectures.
+
+1. CUDA 10.X -- [50, 52, 53] (Maxwell), [60, 61, 62] (Pascal), [70, 72] (Volta), [75] (Turing)
+2. CUDA 11.X -- [50, 52, 53] (Maxwell), [60, 61, 62] (Pascal), [70, 72] (Volta), [75] (Turing), [80] (Ampere)
```
Comment (awthomp) [inline README.md:121]: Let's remove this section. I know what you're trying to say here, but I think the message gets muddled a bit. The big goal, for me, is to:

- Say that Conda is the preferred installation mechanism, and if you use Conda, don't worry about your CC unless you're before Maxwell. Additionally, if you're on Ampere, you'll have to use CUDA 11

Let's address these fine points in a README overhaul. It's probably time for me to do an update anyway :).

Comment (mnicely) [reply README.md:121]: Remove with no replacement?

Comment (mnicely) [reply README.md]: Back to original

**Review by awthomp — APPROVED** [review]

**Pull request merged**